### PR TITLE
Make sure that direct resource load path is only executed for synchro…

### PR DESCRIPTION
**Pull request by jmarchionatto**
…nous searches

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java [hunk 1]
```diff
@@ -361,13 +361,13 @@ public IBundleProvider registerSearch(final IFhirResourceDao<?> theCallingDao, f
 //		SearchStrategyFactory.ISearchStrategy searchStrategy = mySearchStrategyFactory.pickStrategy(theResourceType, theParams, theRequestDetails);
 //		return searchStrategy.get();
 
-		if (mySearchStrategyFactory.isSupportsHSearchDirect(theResourceType, theParams, theRequestDetails)) {
-			ourLog.info("Search {} is using direct load strategy", searchUuid);
-			SearchStrategyFactory.ISearchStrategy direct =  mySearchStrategyFactory.makeDirectStrategy(searchUuid, theResourceType, theParams, theRequestDetails);
-			return direct.get();
-		}
-
 		if (theParams.isLoadSynchronous() || loadSynchronousUpTo != null || isOffsetQuery) {
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java:364]: Please move this line inside the strategy calculation.  That's the whole point of the strategy factory -- avoiding the boolean checks being distributed over the entire trace.

**Review by michaelabuckley — CHANGES_REQUESTED** [review]
Looks ok, but I'd like to consolidate the logic.

Have your verified that we produce next links on the non-offset path?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java (hunk 1 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java:364]: I don't understand your ask, as we still want to have a path for not-direct synchronous query, so the check would be duplicated unless we left it outside of the strategy.

**Review by michaelabuckley — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java (hunk 1 above)
Comment (jmarchionatto) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java:364]: We agreed on look into this change by deepening the scope of strategy sometime later

**Pull request merged**